50_wa_diff_overdose.py

```diff
@@ -0,0 +1,299 @@
+        return 0
+
+    if record == "Drug poisonings (overdose) Unintentional (X40-X44)":
+        return 1
+
+    if record == "Drug poisonings (overdose) Suicide (X60-X64)":
+        return 1
+
+    if record == "Drug poisonings (overdose) Undetermined (Y10-Y14)":
+        return 1
+
+    else:
+        return "error"
+
+
+# copy to fix the dreaded "A value is trying to be set on a copy of a slice" error
+wa_deaths = washington.copy()
+comp_deaths = comp.copy()
+
+# apply new function to our df
+wa_deaths["overdose"] = wa_deaths["Drug/Alcohol Induced Cause"].apply(
+    lambda x: select_overdose(x)
+)
+comp_deaths["overdose"] = comp_deaths["Drug/Alcohol Induced Cause"].apply(
+    lambda x: select_overdose(x)
+)
+
+# filter accordingly based on new column
+wa_deaths = wa_deaths[wa_deaths["overdose"] != 0]
+comp_deaths = comp_deaths[comp_deaths["overdose"] != 0]
+
+wa_deaths["overdose_per_100k"] = wa_deaths["Deaths"] / wa_deaths["Population"] * 100_000
+comp_deaths["overdose_per_100k"] = (
+    comp_deaths["Deaths"] / comp_deaths["Population"] * 100_000
+)
+
+wa_result = (
+    wa_deaths.groupby(["Year", "County"])["overdose_per_100k"].sum().reset_index()
+)
+comp_result = (
+    comp_deaths.groupby(["Year", "StateName", "CountyName"])["overdose_per_100k"]
+    .sum()
+    .reset_index()
+)
+comp_result[comp_result["StateName"] == "Oregon"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Idaho"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Montana"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Nevada"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Wyoming"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+wa_result.groupby("Year")["overdose_per_100k"].mean()
+comp_result.groupby("Year")["overdose_per_100k"].mean()
+wa_result = wa_result.groupby("Year")["overdose_per_100k"].mean().reset_index()
+
+comp_result = comp_result.groupby(["Year"])["overdose_per_100k"].mean().reset_index()
+# create a scale for number of years before and after 2012 (target year)
+
+
+def scale_years(year):
+    if year == 2009:
+        return -3
+    if year == 2010:
+        return -2
+    if year == 2011:
+        return -1
+    if year == 2012:
+        return 0
+    if year == 2013:
+        return 1
+    if year == 2014:
+        return 2
+    if year == 2015:
+        return 3
+
+
+wa_result["year relative to policy"] = wa_result["Year"].apply(lambda x: scale_years(x))
+comp_result["year relative to policy"] = comp_result["Year"].apply(
+    lambda x: scale_years(x)
+)
+
+# double check no nulls in "year relative to policy"
+
+assert wa_result["year relative to policy"].isnull().sum() == 0
+assert comp_result["year relative to policy"].isnull().sum() == 0
+
+# doing this in case the float == int comparison causes issues
+# split into before 2007 and after 2007
```

Comment:
As suggested in lines 101 - 124, this code needs to be deleted, since we are doing away with the column - "year relative to policy".

And just a general tip for lines 131 - 132, it is always good to check for the edge cases, as you have done here, even if it may not be required, after we do away with the column, "year relative to policy." But the easy way to do so is that as soon as you load the dataset, you should first and foremost look at the attributes and observations before you go any further, as I have said so in my comments after line 11. This makes your analysis easier and saves you the trouble of worrying about whether the data is clean or valid or not.

Reply:
Thank you for mentioning the tips for me!